lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+    }
+  }
+}
+
+class BCNLCommand extends BCNLMessage {
+  constructor(command) {
+    if (typeof command === 'number') {
+      // For the ASCII control messages
+      super(Buffer.from([command]));
+    } else {
+      // For regular strings
+      super(Buffer.from(command, 'ascii'));
+    }
+  }
+
+  // TODO - should we do getResponse differently?
+  // Perhaps a member in BNCLRequest that points to a returning class?
+  // eslint-disable-next-line class-methods-use-this
+  getResponse(payload) {
+    return new BCNLCommandResponse(payload);
+  }
+}
+
+class DeviceInfoRequestResponse extends BCNLMessage {
+  getModelAndSerial() {
+    const serialMatch = /\d{4}-\d{7}/.exec(this.payload);
+    return serialMatch.length === 1 ? serialMatch[0] : null;
+  }
+}
+
+class DeviceInfoRequestCommand extends BCNLCommand {
+  constructor() {
+    super('X');
+  }
+
+  // Override send(), because we do a 'double read' after sending the request
+  send(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS,
+    sendDelay = BCNLMessage.SEND_DELAY_MS) {
+    // eslint-disable-next-line no-unused-vars
+    return new Promise((resolve, reject) => {
+      let response1 = null;
+
+      // Timeout might be zero, but the callback will fire anyway
+      // We use sendMessage instead of super.send() because we want to pull raw
+      // data only, and we can't determine response types until after we've checked
+      // them. The CNL can send the messages in different orders.
+      this.sendMessage(hidDevice)
+        .then(() => BCNLMessage.wait(sendDelay))
+        .then(() => this.readMessage(hidDevice, readTimeout))
+        .then((response) => {
+          response1 = response;
+        })
+        .then(() => BCNLMessage.wait(sendDelay))
+        .then(() => this.readMessage(hidDevice, readTimeout))
+        .then((response2) => {
+          let astmInfo = '';
+
+          if (response1[0] === BCNLMessage.ASCII_CONTROL.EOT) {
+            astmInfo = Buffer.from(response1).toString('ascii');
+            new BCNLCommandResponse(response2).checkAsciiControl(BCNLMessage.ASCII_CONTROL.ENQ);
+          } else {
+            astmInfo = Buffer.from(response2).toString('ascii');
+            new BCNLCommandResponse(response1).checkAsciiControl(BCNLMessage.ASCII_CONTROL.ENQ);
+          }
+          const message = new DeviceInfoRequestResponse(astmInfo);
+          resolve(message);
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+  }
+}
+
+class MinimedPumpSession {
+  constructor(modelAndSerial) {
+    this.envelopeSequenceNumber = 0;
+    this.telDSequenceNumber = 0;
+    this.comDSequenceNumber = 0;
+    this.modelAndSerial = modelAndSerial;
+    this.radioChannel = null;
+    this.linkMAC = null;
+    this.pumpMAC = null;
+    this.key = null;
+  }
+
+  get serialNumber() {
+    return this.modelAndSerial.replace(/\d+-/, '');
+  }
+
+  get iv() {
```

Comment:
IV is a standard abbreviation in cryptography, in a similar way that we commonly use cgm, sg, bg, etc.
See https://en.wikipedia.org/wiki/Initialization_vector and https://nodejs.org/api/crypto.html#crypto_crypto_createcipheriv_algorithm_key_iv